src/main/java/org/terasology/gooeysQuests/quests/fortressvillage/FortressVillageQuestSystem.java

```diff
@@ -0,0 +1,223 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.gooeysQuests.quests.fortressvillage;
+
+import org.terasology.assets.management.AssetManager;
+import org.terasology.entitySystem.entity.EntityBuilder;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.BeforeDeactivateComponent;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.gooeysQuests.api.*;
+import org.terasology.logic.inventory.InventoryManager;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.Side;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.registry.In;
+import org.terasology.structureTemplates.events.CheckSpawnConditionEvent;
+import org.terasology.structureTemplates.events.SpawnStructureEvent;
+import org.terasology.structureTemplates.interfaces.BlockPredicateProvider;
+import org.terasology.structureTemplates.interfaces.StructureTemplateProvider;
+import org.terasology.structureTemplates.util.BlockRegionTransform;
+import org.terasology.world.WorldProvider;
+import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockManager;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.function.Predicate;
+
+/**
+ * Makes gooey offer a fortress village quest
+ */
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class FortressVillageQuestSystem extends BaseComponentSystem {
+
+    private static final int MAX_HORIZONTAL_DISTANCE = 20;
+    private static final int VERTICAL_SCAN_DISTANCE = 5;
+    @In
+    private AssetManager assetManager;
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private WorldProvider worldProvider;
+
+    @In
+    private BlockManager blockManager;
+
+    @In
+    private InventoryManager inventoryManager;
+
+    @In
+    private BlockPredicateProvider blockPredicateProvider;
+
+    @In
+    private StructureTemplateProvider structureTemplateProvider;
+
+    private Map<EntityRef, FoundSpawnPossiblity> questToFoundSpawnPossibilityMap = new HashMap<>();
+
+    private Random random = new Random();
+
+    private Predicate<Block> isAirCondition;
+    private Predicate<Block> isGroundCondition;
+
```

Comment:
Occasional double empty lines like these don't really add anything, one blank line is enough :-)